### PR TITLE
Trigger Phase Improvements

### PR DESCRIPTION
**Pull request by iamalisajid**
#### Summary of changes
**Resolves:** #4  

- Asterisks are not must for the trigger phase.
- Added examples for trigger phrase with and without special chracters.
- Fixed the wrong usage link in readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ This action integrates asana with github.
 
 ### `trigger-phrase`
 
-**Required** Prefix before the task i.e ASANA TASK: https://app.asana.com/1/2/3/.
+**Required** Prefix before the task i.e ASANA TASK: https://app.asana.com/1/2/3/. For special chracters in the trigger phrase refer to the examples.
```
Comment (gshkolnik) [inline README.md]:
```suggestion
**Required** Prefix before the task i.e ASANA TASK: https://app.asana.com/1/2/3/. For special characters in the trigger phrase refer to the examples.
```

File: README.md [hunk 2]
```diff
@@ -31,13 +31,27 @@ targets: '[{"project": "Backlog", "section": "Development Done"}, {"project": "C
...
-uses: https://github.com/insurify/github-actions@v2.0.0
+uses: insurify/github-asana-action@v3.0.0
 with:
   asana-pat: 'Your PAT'
   task-comment: 'View Pull Request Here: '
   trigger-phrase: 'Asana Task:'
   targets: '[{"project": "Backlog", "section": "Development Done"}, {"project": "Current Sprint", "section": "In Review"}]'
+```
+
+#### With special chracters:
```
Comment (gshkolnik) [inline README.md]:
```suggestion
#### With special characters:
```

File: README.md [hunk 3]
```diff
@@ -31,13 +31,27 @@ targets: '[{"project": "Backlog", "section": "Development Done"}, {"project": "C
...
 
-## Example usage
+## Examples
+
+
+#### Without special chracters:
 
 ```yaml
-uses: https://github.com/insurify/github-actions@v2.0.0
+uses: insurify/github-asana-action@v3.0.0
```
Comment (gshkolnik) [inline README.md]: I don't think you need to bump your major version. The change you're making is minor, and it's also backward compatible. All the existing patterns will continue working.

**Review by gshkolnik — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -31,13 +31,27 @@ targets: '[{"project": "Backlog", "section": "Development Done"}, {"project": "C
...
+```
+
+#### With special chracters:
+
+```yaml
+uses: insurify/github-asana-action@v3.0.0
+with:
+  asana-pat: 'Your PAT'
+  task-comment: 'View Pull Request Here: '
+  trigger-phrase: "\\*\\*Asana Task:\\*\\*"
```
Comment (srehaider) [inline README.md:58]: We can use [escape-string-regexp](https://www.npmjs.com/package/escape-string-regexp) or main [function](https://github.com/sindresorhus/escape-string-regexp/blob/master/index.js#L3) from it to convert any string to valid escaped regular expression string, so end-user don't need to go into the hassle of writing a valid regular expression.

File: index.js [hunk 5]
```diff
@@ -50,7 +50,7 @@ try {
     TASK_COMMENT = core.getInput('task-comment'),
     PULL_REQUEST = github.context.payload.pull_request,
     REGEX = new RegExp(
-      `\\*\\*${TRIGGER_PHRASE}\\*\\* \\[(.*?)\\]\\(https:\\/\\/app.asana.com\\/(\\d+)\\/(?<project>\\d+)\\/(?<task>\\d+).*?\\)`,
+      `${TRIGGER_PHRASE} \\[(.*?)\\]\\(https:\\/\\/app.asana.com\\/(\\d+)\\/(?<project>\\d+)\\/(?<task>\\d+).*?\\)`,
```
Comment (srehaider) [inline index.js]: We can also make a space after `TRIGGER_PHRASE` optional or add it to trigger phrase to make it more flexible. Adding a sample of PR description will make it more clear for users, how to use it properly.

File: README.md (hunk 3 above)
Comment (iamalisajid) [reply README.md]: If anyone wants to try the trigger phrase without asterisk in the description we won't be able to catch it as we have hardcoded the asterisk and a space in the previous versions. 
Though we can release a minor version of this release, thoughts ?

File: index.js (hunk 5 above)
Comment (iamalisajid) [reply index.js]: https://github.com/actions/toolkit/issues/339

File: README.md [hunk 6]
```diff
@@ -30,14 +30,18 @@ targets: '[{"project": "Backlog", "section": "Development Done"}, {"project": "C
 ```
 if you don't want to move task omit `targets`.
 
-
+## Sample PR Description
+``
+**Asana Task:** [Task Name](https://app.asana.com/0/1/2)
+``
+*Always leave a blank space between the trigger phrase & the asana url.*
```
Comment (gshkolnik) [inline README.md]: why is this needed?

File: README.md (hunk 3 above)
Comment (gshkolnik) [reply README.md]: sure, minor release is fine.

File: index.js (hunk 5 above)
Comment (gshkolnik) [reply index.js]: I don't understand your comment, @iamalisajid . I see that there is an issue with leading spaces, but the way I understand this, it's not an issue here, correct?

Comment (gshkolnik) [reply index.js]: I also do agree, that you can simply update this REGEX to allow for zero or more spaces between the trigger phrase and the URL.

Comment (iamalisajid) [reply index.js]: Actually currently github have a issue logged that they remove leading spaces from the inputs as they are using `.trim()`. 
Due to which if the users event put any leading spaces in the trigger phrase it'll be ignored.

File: README.md (hunk 6 above)
Comment (iamalisajid) [reply README.md:37]: Due to that particular issue, mentioned above.

File: index.js (hunk 5 above)
Comment (iamalisajid) [reply index.js]: Though, I've made the space optional using regex.

File: README.md (hunk 6 above)
Comment (iamalisajid) [reply README.md]: Removed after making the space optional.

File: README.md [hunk 7]
```diff
@@ -30,14 +30,18 @@ targets: '[{"project": "Backlog", "section": "Development Done"}, {"project": "C
...
+## Sample PR Description
+``
+**Asana Task:** [Task Name](https://app.asana.com/0/1/2)
+``
 
 ## Example usage
 
 ```yaml
-uses: https://github.com/insurify/github-actions@v2.0.0
+uses: insurify/github-asana-action@v2.1.0
```
Comment (gshkolnik) [inline README.md]:
```suggestion
uses: insurify/github-asana-action@v2.0.1
```

File: index.js [hunk 8]
```diff
@@ -46,11 +46,12 @@ async function asanaOperations(
 try {
   const ASANA_PAT = core.getInput('asana-pat'),
     TARGETS = core.getInput('targets'),
-    TRIGGER_PHRASE = core.getInput('trigger-phrase'),
+    TRIGGER_PHRASE = core.getInput('trigger-phrase').replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
```
Comment (gshkolnik) [inline index.js]: I'm not sure I understand what's happening here. I don't think we need it.

Comment (srehaider) [reply index.js]: `|\{}()[]^$+*?.` are special characters in the regular expression and have some specific meaning. To detect them as they are, we need to escape them. `*` means optional or many preceding character or group, to detect an `*` we will have to escape it like `\*`. If we will remove this change end-user will be required to pass a valid regular expression string to detect `TRIGGER_PHRASE` correctly.

Comment (gshkolnik) [reply index.js]: I understand that you're trying to be helpful, but I think you might be actual making things more difficult to the users. If they want to have special characters in their phrase, they can figure out themselves how to escape them. However, by doing what you're proposing you're making it impossible for them to have regular expressions. So why wouldn't we want for them to be able to do that?

File: README.md (hunk 7 above)
Comment (iamalisajid) [reply README.md]: @gshkolnik if i'm not wrong the last number represents the patch and I don't think we are patching something in release 2, thoughts ?

Comment (gshkolnik) [reply README.md]: Yes, what we're doing is a simple, backward-compatible bug fix or in other words a patch. We are certainly not doing a major release here.
I did get the version wrong though:
```suggestion
uses: insurify/github-asana-action@v1.0.1
```

File: index.js (hunk 8 above)
Comment (srehaider) [reply index.js]: My initial understanding was `TRIGGER_PHRASE` will be a static string which will not change across different PRs and we don't need a regular expression to detect a static string. Surely we can allow users to add their own regular expression and it will improve flexibility too. It will be great to describe this part well in the readme so users don't need to tinker around it.

File: README.md (hunk 7 above)
Comment (iamalisajid) [reply README.md]: @gshkolnik the current code have one extra thing of `TARGETS` so if we add this to `1.0.1` targets will also be added to this version, so I think it's better to go with `2.0.1` or `2.1.0`.

File: index.js (hunk 8 above)
Comment (iamalisajid) [reply index.js]: I think let's do what @gshkolnik  suggests here as it will be more flexible for user, alongside we can add details in read me that how you can extend the regex pattern.

File: README.md (hunk 7 above)
Comment (gshkolnik) [reply README.md]: let's split the difference and go with `1.1.0` - sound good? :)

Comment (srehaider) [reply README.md]: Sorry, we are unable to explain it, we already have marked the last version as `2.0.0` so moving from `2.0.0` to `1.1.0` doesn't make sense.

Comment (gshkolnik) [reply README.md]: I thought the current version was 1.0. At least this is what we're using in ensurify repo.

Comment (iamalisajid) [reply README.md]: Yes you're right, we haven't updated the version in our repos.

Comment (iamalisajid) [reply README.md]: @gshkolnik so are we good with `2.1.0` ?

Comment (gshkolnik) [reply README.md]: @iamalisajid  - no, this change does not amount to a major release. My preference is that we actually don't change the version at all really. However, if you feel that we need to, we can go with `1.0.1` or maybe the highest as `1.1.0`.

**Review by gshkolnik — APPROVED** [review]
